applications/StatisticsApplication/custom_python/add_custom_spatial_methods_to_python.cpp

```diff
@@ -0,0 +1,181 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya (https://github.com/sunethwarna)
+//
+
+// System includes
+
+// External includes
+#include <pybind11/pybind11.h>
+
+// Project includes
+
+// Application includes
+#include "custom_methods/spatial_methods.h"
+
+// Include base h
+#include "custom_python/add_custom_spatial_methods_to_python.h"
+
+// MPI methods doesn't support vector and matrix SumAll methods,
+// therefore only double and fixed array templates are added
+#ifndef ADD_KRATOS_STATISTICS_SPATIAL_CONTAINER_VALUE_METHOD_PYTHON_INTERFACE
+#define ADD_KRATOS_STATISTICS_SPATIAL_CONTAINER_VALUE_METHOD_PYTHON_INTERFACE(         \
```

Comment:
are you sure you need a macro for this
This is e.g. how I do it: https://github.com/KratosMultiphysics/Kratos/blob/master/applications/MappingApplication/custom_python/add_custom_mappers_to_python.cpp#L134-L167

Reply:
The problem for me is, i have permutations as in, The same method needs to be exposed to python which is coming from templated different classes for different containers. :/